scipy/stats/_qmc.py

```diff
@@ -567,6 +567,9 @@ class QMCEngine(ABC):
 
     @abstractmethod
     def __init__(self, d, seed=None):
+        if not isinstance(d, int):
```

Comment:
```suggestion
        if not np.issubdtype(type(d), np.integer):
```
I think this would be a little more flexible, allowing e.g. `np.int64`.

Reply:
Ah ok sure, thanks!